staging/src/k8s.io/apimachinery/pkg/util/sets/set.go

```diff
@@ -177,61 +172,28 @@ func (s1 Set[T]) Equal(s2 Set[T]) bool {
 	return len(s1) == len(s2) && s1.IsSuperset(s2)
 }
 
-type sortableSlice[T SetTypes] struct {
-	slice []T
-	OrderedFuncs[T]
-}
-
-func (s sortableSlice[T]) Len() int {
-	if s.LenFunc == nil {
-		return len(s.slice)
-	}
-	return s.LenFunc(s.slice)
-}
-func (s sortableSlice[T]) Less(i, j int) bool {
-	if s.LessFunc == nil {
-		// unsorted if no less func specified
-		return true
-	}
-	return s.LessFunc(s.slice, i, j)
-}
-func (s sortableSlice[T]) Swap(i, j int) {
-	if s.SwapFunc == nil {
-		s.slice[i], s.slice[j] = s.slice[j], s.slice[i]
-		return
+// List returns the slice with contents in random order.
+func (s Set[T]) List() []T {
+	res := make([]T, 0, len(s))
+	for key := range s {
+		res = append(res, key)
 	}
-	s.SwapFunc(s.slice, i, j)
-}
-
-// OrderedFuncs allows custom sorting methods like sort.Interface.
-// List will be unsorted if LessFunc absent, in other word, UnsortedList
-// should be used in this scenario.
-type OrderedFuncs[T SetTypes] struct {
-	LenFunc  func(s []T) int
-	LessFunc func(s []T, i, j int) bool
-	SwapFunc func(s []T, i, j int)
+	return res
 }
 
-// List returns the contents as a sorted T slice by given OrderedFuncs.
-func (s Set[T]) List(orderedFuncs OrderedFuncs[T]) []T {
-	res := sortableSlice[T]{
-		slice:        make([]T, 0, len(s)),
-		OrderedFuncs: orderedFuncs,
-	}
-	for key := range s {
-		res.slice = append(res.slice, key)
-	}
-	sort.Sort(res)
-	return res.slice
+// SortedList returns the contents as a sorted T slice by given less func.
+func (s Set[T]) SortedList(less func(slice []T, i, j int) bool) []T {
```

Comment:
If you want to have a function instead of a separate function taking constraints.Ordered, at least make the function signature be `func(a T, b T) bool`? I think I prefer the solution with the separate function with an ordered constraint, though.

Reply:
I choose the separate function taking constraints.Ordered either. :)